### DISCUSSION TITLE
Trains Pathfinding without turnaround

### DISCUSSION THREAD

**Discussion by Schety** [voxeltycoon/issues#365] [General]
**Describe the bug**
"allow Train turnaround" off
It seems that if there's a way forward, the Train will always chose that one no matter how much longer that way is and regardles of the order i give him.
it will drive a loong way trough a loop, rather than to the Waypoint acoupple of meters behind it.

if the Train first goes to the Waypont (if the waypoint is in a dead end) it works (obviously)

A Drive Backwards until next Station/waypoint would maybe still be a good addition.

(I do know that's a rather specific "problem")

**Build version**
0.85.0.24 steam

**Platform**
Windows 10

**Reproducibility**
every time

**Comment by victorleblais** [comment]
I don't think it's a bug but simply a design choice: trains will always prefer driving forward instead of reversing.

What you're describing is more of a suggestion to have manual control over the train pathfinding.

**Comment by victorleblais** [comment]
Here's a screenshot to illustrate:
![image](https://user-images.githubusercontent.com/4365075/115159435-5fa1ad80-a093-11eb-89a6-3aec195389f5.png)

There could be an order command to "drive backwards", which could be added the end of the A order.

**Comment by Schety** [comment]
That would probably be a good addition. but if i give the Manual "turn around" order it will just turn around 2 seconds later which could give some slight Problems later when Trains don't teleport to Depot any more, If for example two Trains stop across from each other because of false signaling.
The trains will just refuse to Drive backwards unless I delet some Rails in front of the train.
I would expect the train to move Backwards a least until the next false Signal (if the "Ignore closst Signal" is off)

(btw. your settup should not work because the Train turns around as soon as the first Waggon is on the Waypoint and not the Lok.The waypoint would need to be farther back)
